src/api/v4/layer/layer.js

```diff
@@ -275,6 +283,13 @@ function _checkSource (source) {
     throw new TypeError('The given object is not a valid source. See "carto.source.Base"');
   }
 }
+
+/**
+ * Return true when a windshaft error is because a styling error.
+ */
+function _isStyleError (windshaftError) {
+  return windshaftError.message && windshaftError.message.indexOf('style') >= 0;
```

Comment:
If they style error messages start with `style` we could check that `windshaftError.message.indexOf('style') === 0` to be more accurate.